src/openloco/platform/platform.posix.cpp

```diff
@@ -21,91 +21,95 @@ int main(int argc, const char** argv)
-    if (path.empty())
+    static fs::path get_home_directory()
     {
-        path = get_home_directory();
-        if (path.empty())
+        auto pw = getpwuid(getuid());
+        if (pw != nullptr)
         {
-            path = "/";
+            return pw->pw_dir;
         }
         else
         {
-            path = path / fs::path(".config");
+            return GetEnvironmentVariable("HOME");
         }
     }
-    return path / fs::path("OpenLoco");
-}
+
+    fs::path get_user_directory()
+    {
+        auto path = fs::path(GetEnvironmentVariable("XDG_CONFIG_HOME"));
+        if (path.empty())
+        {
+            path = get_home_directory();
+            if (path.empty())
+            {
+                path = "/";
+            }
+            else
+            {
+                path = path / fs::path(".config");
+            }
+        }
+        return path / fs::path("OpenLoco");
+    }
 #endif
 
 #if !(defined(__APPLE__) && defined(__MACH__))
-fs::path openloco::platform::GetCurrentExecutablePath()
-{
-    char exePath[PATH_MAX] = { 0 };
-#ifdef __linux__
-    auto bytesRead = readlink("/proc/self/exe", exePath, sizeof(exePath));
-    if (bytesRead == -1)
+    fs::path GetCurrentExecutablePath()
     {
-        console::error("failed to read /proc/self/exe");
-    }
+        char exePath[PATH_MAX] = { 0 };
+#ifdef __linux__
+        auto bytesRead = readlink("/proc/self/exe", exePath, sizeof(exePath));
+        if (bytesRead == -1)
+        {
+            console::error("failed to read /proc/self/exe");
+        }
 #elif defined(__FreeBSD__)
-    const int32_t mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
-    auto exeLen = sizeof(exePath);
-    if (sysctl(mib, 4, exePath, &exeLen, nullptr, 0) == -1)
-    {
-        console::error("failed to get process path");
-    }
+        const int32_t mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+        auto exeLen = sizeof(exePath);
+        if (sysctl(mib, 4, exePath, &exeLen, nullptr, 0) == -1)
+        {
+            console::error("failed to get process path");
+        }
 #elif defined(__OpenBSD__)
-    // There is no way to get the path name of a running executable.
-    // If you are not using the port or package, you may have to change this line!
-    strlcpy(exePath, "/usr/local/bin/", sizeof(exePath));
+        // There is no way to get the path name of a running executable.
+        // If you are not using the port or package, you may have to change this line!
+        strlcpy(exePath, "/usr/local/bin/", sizeof(exePath));
 #else
 #error "Platform does not support full path exe retrieval"
-#endif
-    return exePath;
-}
+#endif // __linux__
+        return exePath;
+    }
+
+    std::string prompt_directory(const std::string& title)
+    {
+        std::string input;
+        std::cout << "Type your Locomotion path: ";
+        std::cin >> input;
+        return input;
+    }
+#endif // !(defined(__APPLE__) && defined(__MACH__))
 
```

Comment:
```suggestion

```

CI's clang-format wants this line gone. My local version appears to be fine with it, though.